app.py

```diff
@@ -14,22 +14,59 @@
 
 @app.route('/')
 def home() -> str:
+    """
+    Returns:
+        str
+    """    
     return render_template('home.html')
 
 
+@app.route('/health')
+def health() -> Response:
+    """
+    Returns:
+        Response
+    """    
+    return jsonify({'message': 'OK'}), 200
+
+
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
-    animals = Animal.query.all()
-    return jsonify({"animals": [AnimalResponse.model_validate(animal).model_dump(mode='json') for animal in animals]})
+    """
+    Returns:
+        Response
+    """    
+    search_value = request.args.get('search')
+
+    if search_value:
+        animals = Animal.query.filter_by(name=search_value).all()
```

Comment:
you can filter by substring like `Model.query.filter(Model.columnName.contains('sub_string'))`